internal/service/backupcompliancepolicy/data_source_backup_compliance_policy.go

```diff
@@ -44,8 +44,7 @@ func DataSource() *schema.Resource {
 			},
 			"on_demand_policy_item": {
 				Type:     schema.TypeList,
-				MaxItems: 1,
-				Optional: true,
+				Computed: true,
```

Comment:
any potential breaking changes ? or plan change detections?
I am asking to make sure you've done some extra testing.

Reply:
while I do not see a risk in this change as [project_id is documented as the only argument](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/backup_compliance_policy#argument-reference) for the data source, we should consider also fixing inconsistencies present in `policy_item_hourly`, `policy_item_daily`, `policy_item_weekly` and `policy_item_monthly` which are not defined as computed. Error seems to only be raised in `on_demand_policy_item` because all nested attributes are computed.

Reply:
@AgustinBettati All the `policy_item_*` fields have a few required parameters such as `retention_unit` and `retention_value` which is the reason why we are not getting the error for them.

Reply:
@marcosuma I was able to test locally that this is not a breaking change.
Testing scenario:

- Run the following configuration
```hcl
resource "mongodbatlas_project" "atlas-project" {
  name   = "test-andrea"
  org_id = "60ddf55c27a5a20955a707d7"
}

resource "mongodbatlas_cluster" "my_cluster" {
  project_id   = mongodbatlas_project.atlas-project.id
  name         = "clusterTest"


  //Provider Settings "block"
  provider_name               = "AWS"
  provider_region_name        = "EU_CENTRAL_1"
  provider_instance_size_name = "M10"
  cloud_backup                = true // enable cloud backup snapshots
}

resource "mongodbatlas_cloud_backup_schedule" "test" {
  project_id   = mongodbatlas_cluster.my_cluster.project_id
  cluster_name = mongodbatlas_cluster.my_cluster.name

  reference_hour_of_day    = 3
  reference_minute_of_hour = 45
  restore_window_days      = 4

  // This will now add the desired policy items to the existing mongodbatlas_cloud_backup_schedule resource
  policy_item_hourly {
    frequency_interval = 1        #accepted values = 1, 2, 4, 6, 8, 12 -> every n hours
    retention_unit     = "days"
    retention_value    = 7
  }
  policy_item_daily {
    frequency_interval = 1        #accepted values = 1 -> every 1 day
    retention_unit     = "days"
    retention_value    = 7
  }
  policy_item_weekly {
    frequency_interval = 1        # accepted values = 1 to 7 -> every 1=Monday,2=Tuesday,3=Wednesday,4=Thursday,5=Friday,6=Saturday,7=Sunday day of the week
    retention_unit     = "weeks"
    retention_value    = 4
  }
  policy_item_monthly {
    frequency_interval = 1        # accepted values = 1 to 28 -> 1 to 28 every nth day of the month  
                                  # accepted values = 40 -> every last day of the month
    retention_unit     = "months"
    retention_value    = 12
  }

}

data "mongodbatlas_cloud_backup_schedule" "test" {
  project_id   = mongodbatlas_cloud_backup_schedule.test.project_id
  cluster_name = mongodbatlas_cloud_backup_schedule.test.cluster_name
}
```

- Run TF apply with the latest provider
```hcl
Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
```
- Updated the provider to use local version and run again terraform apply
```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

Reply:
Just to be sure, is it possible that the relevant data source `backup_compliance_policy` was not tested in the above test configuration?

> All the policy_item_* fields have a few required parameters such as retention_unit and retention_value which is the reason why we are not getting the error for them

Correct, I was bringing this up as I don't see why there should be optional or required attributes given that they should not be defined in the configuration of the data source. All of them should be computed except for `project_id` which is the only argument for using the data source.

Reply:
> Just to be sure, is it possible that the relevant data source backup_compliance_policy was not tested in the above test configuration?

ops, I copied/pasted the wrong conf 🤦 I updated my message. Thanks!

> Correct, I was bringing this up as I don't see why there should be optional or required attributes given that they should not be defined in the configuration of the data source. All of them should be computed except for project_id which is the only argument for using the data source.

Thanks for the clarification. I agree, I updated the schema to make them computed as well. I also retested the new changes with the TF conf and no changes where detected. Thanks!